src/main/java/at/ac/tuwien/kr/alpha/common/BasicAnswerSetFormatter.java

```diff
@@ -0,0 +1,33 @@
+package at.ac.tuwien.kr.alpha.common;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
+
+import at.ac.tuwien.kr.alpha.common.atoms.Atom;
+
+public class BasicAnswerSetFormatter implements AnswerSetFormatter<String> {
+
+	private final String atomSeparator;
+
+	public BasicAnswerSetFormatter(String atomSeparator) {
+		this.atomSeparator = atomSeparator;
+	}
+
+	@Override
+	public String format(AnswerSet answerSet) {
+		List<String> predicateInstanceStrings = new ArrayList<>();
```

Comment:
There is a lot of `String`-creation and copying going on here for printing an answer set. I thought for a while on how to avoid that, but it already starts at `Atom.toString()`, so in order to get rid of the string-copying, one would need to start there and add facilities to write an `Atom` to a `StringBuilder`.
(Just mentioning this as potential future optimisation.)